tests/test_save_fields.py

```diff
@@ -59,6 +61,28 @@ def test_save_dirty_related_field():
     assert MixedFieldsModelTest.objects.get(pk=tmfm.pk).fkey_id == tm1.id
 
 
+@pytest.mark.django_db
+def test_save_dirty_full_save_on_adding():
+    # import atomic in the local scope as making that in the global scope somehow
```

Comment:
can you rewrite the test to just check that `save_dirty_fields()` works on a unsaved model instance.